crypto/ml_kem/ml_kem.c

```diff
@@ -1672,6 +1705,26 @@ int ossl_ml_kem_encode_key_seed(uint8_t *out, size_t len,
     return 1;
```

Comment:
On line 1704, you dereference `key->z`, but haven't checked that it isn't NULL.  I suspect that's what causes the abortion visible in many CI runs.  [This one](https://github.com/openssl/openssl/actions/runs/12835030831/job/35793585184#step:8:126) comes with the following message:

```
crypto/ml_kem/ml_kem.c:1704:5: runtime error: null pointer passed as argument 2, which is declared to never be null
```

Reply:
Fixed. The seed representation is different when it is a seed-only key.